src/tools/rustdoc-gui/tester.js

```diff
@@ -101,25 +136,77 @@ async function main(argv) {
     let failed = false;
     let files;
     if (opts["files"].length === 0) {
-        files = fs.readdirSync(opts["tests_folder"]).filter(file => path.extname(file) == ".goml");
+        files = fs.readdirSync(opts["tests_folder"]);
     } else {
-        files = opts["files"].filter(file => path.extname(file) == ".goml");
+        files = opts["files"];
+    }
+    files = files.filter(file => path.extname(file) == ".goml");
+    if (files.length === 0) {
+        console.error("rustdoc-gui: No test selected");
+        process.exit(2);
     }
-
     files.sort();
-    for (var i = 0; i < files.length; ++i) {
-        const testPath = path.join(opts["tests_folder"], files[i]);
-        await runTest(testPath, options).then(out => {
-            const [output, nb_failures] = out;
-            console.log(output);
-            if (nb_failures > 0) {
+
+    console.log(`Running ${files.length} rustdoc-gui tests...`);
+    process.setMaxListeners(files.length + 1);
+    let tests = [];
+    let results = {
+        successful: [],
+        failed: [],
+        errored: [],
+    };
+    const status_bar = char_printer(files.length);
+    for (let i = 0; i < files.length; ++i) {
+        const file_name = files[i];
+        const testPath = path.join(opts["tests_folder"], file_name);
+        tests.push(
+            runTest(testPath, options)
+            .then(out => {
+                const [output, nb_failures] = out;
+                results[nb_failures === 0 ? "successful" : "failed"].push({
+                    file_name: file_name,
+                    output: output,
+                });
+                if (nb_failures > 0) {
+                    status_bar.erroneous()
+                    failed = true;
+                } else {
+                    status_bar.successful()
+                }
+            })
+            .catch(err => {
+                results.errored.push({
+                    file_name: file_name,
+                    output: err,
+                });
+                status_bar.erroneous();
                 failed = true;
-            }
-        }).catch(err => {
-            console.error(err);
-            failed = true;
+            })
+        );
+        if (no_headless) {
+            await tests[i];
+        }
+    }
+    await Promise.all(tests);
+    // final \n after the tests
+    console.log("\n");
```

Comment:
Shouldn't it be:

```suggestion
    console.log("");
```

? Or do you specifically want an empty line?

Reply:
If so, maybe better to use two `console.log("")` instead I think. No strong opinion though.

Reply:
I wanted a bit more separation between the sections for readability so the progress bar is visually spaced from the outputs.
The output should be the same on all platforms because we don't write anything on the empty line but fully correct it should be two `console.log("")`, will change that

Reply:
Replaced it with a better finish method that prints the final stats